doc/source/tutorial/stats/sampling_hinv.rst

```diff
@@ -0,0 +1,191 @@
+    >>> rvs2 = rng.rvs(size=100, random_state=np.random.default_rng(seed))
+    >>> np.allclose(rvs1, rvs2)
+    True
+
+To check that the random variates closely follow the given distribution, we can
+look at its histogram:
+
+.. plot::
+
+    >>> import numpy as np
+    >>> from scipy.stats import NumericalInverseHermite
+    >>> from scipy.stats import norm, genexpon
+    >>> from scipy.special import ndtr
+    >>> import matplotlib.pyplot as plt
+    >>> class StandardNormal:
+    ...     def pdf(self, x):
+    ...        return 1/np.sqrt(2*np.pi) * np.exp(-x**2 / 2)
+    ...     def cdf(self, x):
+    ...        return ndtr(x)
+    ... 
+    >>> dist = StandardNormal()
+    >>> urng = np.random.default_rng()
+    >>> rng = NumericalInverseHermite(dist, random_state=urng)
+    >>> rvs = rng.rvs(10000)
+    >>> x = np.linspace(rvs.min()-0.1, rvs.max()+0.1, 1000)
+    >>> fx = norm.pdf(x)
+    >>> plt.plot(x, fx, 'r-', lw=2, label='true distribution')
+    >>> plt.hist(rvs, bins=20, density=True, alpha=0.8, label='random variates')
+    >>> plt.xlabel('x')
+    >>> plt.ylabel('PDF(x)')
+    >>> plt.title('Numerical Inverse Hermite Samples')
+    >>> plt.legend()
+    >>> plt.show()
+
+Given the derivative of the PDF w.r.t the variate (i.e. ``x``), we can use
+quintic Hermite interpolation to approximate the PPF by passing the ``order``
+parameter:
+
+    >>> class StandardNormal:
+    ...     def pdf(self, x):
+    ...        return 1/np.sqrt(2*np.pi) * np.exp(-x**2 / 2)
+    ...     def dpdf(self, x):
+    ...        return -1/np.sqrt(2*np.pi) * x * np.exp(-x**2 / 2)
+    ...     def cdf(self, x):
+    ...        return ndtr(x)
+    ... 
+    >>> dist = StandardNormal()
+    >>> urng = np.random.default_rng()
+    >>> rng = NumericalInverseHermite(dist, order=5, random_state=urng)
+
+Higher orders result is fewer number of intervals:
+
+    >>> rng3 = NumericalInverseHermite(dist, order=3)
+    >>> rng5 = NumericalInverseHermite(dist, order=5)
+    >>> rng3.intervals, rng5.intervals
+    (3000, 522)
+
+The u-error can be estimated by calling the `u_error` method. It runs a small
+Monte-Carlo simulation to estimate the u-error. By default, 100000 samples are
+used. To change this, you can pass the number of samples as an argument:
+
+    >>> rng1 = NumericalInverseHermite(dist, u_resolution=1.e-10)
+    >>> rng1.u_error(sample_size=1000000)  # uses one million samples
+    UError(max_error=9.53167544892608e-11, mean_absolute_error=2.2450136432146864e-11)
+
+This returns a namedtuple which contains the maximum u-error and the mean
+absolute u-error.
+
+The u-error can be reduced by decreasing the u-resolution (maximum allowed u-error):
+
+    >>> rng2 = NumericalInverseHermite(dist, u_resolution=1.e-13)
+    >>> rng2.u_error(sample_size=1000000)
+    UError(max_error=9.32027892364129e-14, mean_absolute_error=1.5194172675685075e-14)
+
+Note that this comes at the cost of increased setup time and number of intervals.
```

Comment:
Is this entirely correct? This phrasing implies that reducing the u-error by decreasing the u-resolution comes at the cost of the number of intervals, but they in fact increase, which seems to be a positive factor in reducing the u-error. 

I think you may have meant that 
```suggestion
Note that this comes at the cost of computation time as a result of the increased setup time and number of intervals. 
```

Reply:
> Is this entirely correct? This phrasing implies that reducing the u-error by decreasing the u-resolution comes at the cost of the number of intervals, but they in fact increase, which seems to be a positive factor in reducing the u-error.

Ah, yes. This makes sense.

> I think you may have meant that

Yes, that's exactly what I meant!

Reply:
Changed it this way, thanks!